### PR TITLE
Add matrix tests

### PR DESCRIPTION
**Pull request by knordbo**
In order to consume this library some test coverage is needed. (#4)
There are more tests that can be added, but its probably a good idea agree how tests are written first. For instance, whats the preference for test names, concise or more full sentence style?

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,8 +11,6 @@ repositories {
 
 dependencies {
     compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
-}
-
-sourceSets {
-    main.kotlin.srcDirs += 'src/main/kotlin'
```
Comment (romainguy) [inline build.gradle]: Why did you remove the sourceSets block?

File: build.gradle [hunk 2]
```diff
@@ -11,8 +11,6 @@ repositories {
...
-}
+    testCompile "org.jetbrains.kotlin:kotlin-test-junit"
+    testCompile "junit:junit:4.12"
+}
```
Comment (romainguy) [inline build.gradle]: Keep an empty line at the end of the file

File: src/test/kotlin/com/curiouscreature/kotlin/math/MatrixTest.kt [hunk 3]
```diff
@@ -0,0 +1,288 @@
+package com.curiouscreature.kotlin.math
```
Comment (romainguy) [inline src/test/kotlin/com/curiouscreature/kotlin/math/MatrixTest.kt]: Add copyright header

File: src/test/kotlin/com/curiouscreature/kotlin/math/MatrixTest.kt [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+
+import org.junit.Assert
+import org.junit.Test
+import kotlin.test.assertEquals
+
+class MatrixTest {
+
```
Comment (romainguy) [inline src/test/kotlin/com/curiouscreature/kotlin/math/MatrixTest.kt]: Remove empty line

File: src/test/kotlin/com/curiouscreature/kotlin/math/MatrixTest.kt [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+        private fun assertArrayEquals(expected: FloatArray, actual: FloatArray, delta: Float = 0.0001f) =
+                Assert.assertArrayEquals(expected, actual, delta)
+
+        /**
+         * @return a FloatArray containing n floats 1f,2f,...,n (float) where n
+         * is the @receiver integer.
+         */
+        private fun Int.floatArray() = FloatArray(this) { (it + 1).toFloat() }
+    }
+
```
Comment (romainguy) [inline src/test/kotlin/com/curiouscreature/kotlin/math/MatrixTest.kt]: Remove empty line

File: build.gradle (hunk 1 above)
Comment (knordbo) [reply build.gradle]: It shouldn't be needed AFAIK as you are using the default convention (https://kotlinlang.org/docs/reference/using-gradle.html#targeting-the-jvm)
I can keep it in though as one could argue it is more declarative.

Comment (romainguy) [reply build.gradle]: I'm fine with removing it then, thanks!
